core/new-gui/src/app/workspace/service/validation/validation-workflow.service.ts

```diff
@@ -0,0 +1,141 @@
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.deletedLink.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+    this.workflowActionService.getTexeraGraph().getOperatorPropertyChangeStream()
+      .subscribe(value => {
+        const operatorID = value.operator.operatorID;
+        const status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+  }
+
+  /**
+   * Gets the observable for operator validation change event.
+   * Contains a boolean variable that indicates:
+   *  - the new status for the validation of operator
+   */
+  public getOperatorValidationStream(): Observable<{status: boolean, operatorID: string}> {
+    return this.operatorValidationStream.asObservable();
+  }
+
+  /**
+   * if all ports of the operator are connected and all properties are completed
+   * then the operator is valid.
+   */
+  public validateOperator(operatorID: string): boolean {
+    if (!this.isOperatorIsolated(operatorID) && this.isJsonSchemaValiad(operatorID)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * This method is used to check whether all required properties of the operator have been completed.
+   * if completed correctly, the operator box is valid.
+   */
+  private isJsonSchemaValiad(operatorID: string): boolean {
+    const operator = this.workflowActionService.getTexeraGraph().getOperator(operatorID);
+    let operatorSchema;
+    let isValid;
+    const ajv = new Ajv ({schemaId: 'auto', allErrors: true});
+    ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
+    if (operator !== undefined) {
+      operatorSchema = this.operatorSchemaList.find(schema => schema.operatorType === operator.operatorType);
+      if (operatorSchema !== undefined) {
+        isValid =  ajv.validate(operatorSchema.jsonSchema, operator.operatorProperties);
+      }
+    }
+
+    if (isValid) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * This method is used to check whether all ports of the operator box has been connected.
+   * if all ports of the operator box is connected, the operator is valid.
+   */
+  private isOperatorIsolated(operatorID: string): boolean {
+     const operator = this.workflowActionService.getTexeraGraph().getOperator(operatorID);
+     let inputPortsNum = null;
+     let outputPortsNum = null;
+     if (operator !== undefined) {
+     inputPortsNum = operator.inputPorts.length;
+     outputPortsNum = operator.outputPorts.length;
+     }
+
+     if (isEqual(inputPortsNum, this.workflowActionService.getTexeraGraph().getInputLinksByOperatorId(operatorID).length) &&
+     isEqual(outputPortsNum, this.workflowActionService.getTexeraGraph().getOutputLinksByOperatorId(operatorID).length)) {
+      return false;
+     }
+     return true;
+  }
```

Comment:
Rather than doing `operator !== undefined`, write the function so that it will throw an error when `operator === undefined`. By doing this, you can detect error from the beginning (since operator should always be defined when validating) and don't need to use `let` for the `inputPortsNum` and `outputPortsNum`.